polyply/src/simple_seq_parsers.py

```diff
@@ -0,0 +1,236 @@
+# limitations under the License.
+from collections import OrderedDict
+from functools import partial
+import json
+import networkx as nx
+from networkx.readwrite import json_graph
+from vermouth.parser_utils import split_comments
+from vermouth.log_helpers import StyleAdapter, get_logger
+
+LOGGER = StyleAdapter(get_logger(__name__))
+
+ONE_LETTER_DNA = {"A": "DA",
+                  "C": "DC",
+                  "G": "DG",
+                  "T": "DT"}
+
+ONE_LETTER_RNA = {"A": "A",
+                  "C": "C",
+                  "G": "G",
+                  "T": "U"}
+
+def _monomers_to_linear_nx_graph(monomers):
+    """
+    Take a list of monomers and generate a linear
+    graph from them setting both the resid and
+    resname attribute.
+    """
+    seq_graph = nx.Graph()
+    mon_range = range(0, len(monomers))
+    seq_graph.add_edges_from(zip(mon_range[:-1], mon_range[1:]))
+    nx.set_node_attributes(seq_graph, {node: resname for node, resname in zip(seq_graph.nodes, monomers)}, "resname")
+    nx.set_node_attributes(seq_graph, {node: node+1 for node in seq_graph.nodes}, "resid")
+    return seq_graph
+
+def parse_plain_delimited(filehandle, delimiter=" "):
+    """
+    Parse a plain delimited text file. The delimiter can
+    be any special character.
+    """
+    with open(filehandle) as file_:
+        lines = file_.readlines()
+
+    monomers = []
+    for line in lines:
+        for resname in line.strip().split(delimiter):
+            monomers.append(resname)
+    seq_graph =  _monomers_to_linear_nx_graph(monomers)
+    return seq_graph
+
+parse_csv = partial(parse_plain_delimited, delimiter=',')
+parse_txt = parse_plain_delimited
+
+def parse_plain(lines, DNA=False, RNA=False):
+    """
+    Parse a plain one letter sequence block either for DNA, RNA,
+    or amino-acids. Lines can be a list of strings or a string.
+    This function also sets the appropiate defaults for the termini
+    of DNA and RNA
+
+    For the format see here:
+
+    https://www.animalgenome.org/bioinfo/resources/manuals/seqformats
+
+    Parameters
+    ----------
+    lines: `abc.iteratable`
+        list of strings matching one letter code DNA, RNA, or AAs
+    DNA: bool
+        if the sequence matches DNA
+    RNA: bool
+        if the sequence matches RNA
+
+    Returns
+    -------
+    `:class:nx.Graph`
+        the plain residue graph matching the sequence
+    """
+    monomers = []
+    for line in lines:
+        for token in line.strip():
+            if token in ONE_LETTER_DNA and DNA:
+                resname = ONE_LETTER_DNA[token]
+            elif token in ONE_LETTER_RNA and RNA:
+                resname = ONE_LETTER_RNA[token]
+            else:
+                msg = f"Cannot find one letter residue match for {token}"
+                raise IOError(msg)
+
+            monomers.append(resname)
+
+    # make sure to set the defaults for the DNA terminals
+    if DNA:
```

Comment:
added the RNA support. This was a leftover since previously it would also parse AA one letter sequences. Since I dropped support for that there is no need to check if RNA or DNA is set. One can just always adding the ending by default.